sky/server/uvicorn.py

```diff
@@ -186,16 +186,17 @@ def _wait_requests(self) -> None:
 
     def interrupt_request_for_retry(self, request_id: str) -> None:
```

Comment:
Why don't we use _kill_requests for this?
I think the change here is subject to the same possible race as the kill_requests change.